src/aap_eda/services/auth.py

```diff
@@ -47,5 +53,52 @@ def create_jwt_token() -> tuple[str, str]:
     if new:
         user.set_unusable_password()
         user.save(update_fields=["password"])
-    rf = RefreshToken.for_user(user)
-    return (str(rf.access_token), str(rf))
+
+    access_token = jwt_access_token(user.id)
+    refresh_token = jwt_refresh_token(user.id)
+    return (access_token, refresh_token)
+
+
+def jwt_access_token(user_id: int) -> str:
+    expiration = timedelta(minutes=settings.JWT_ACCESS_TOKEN_LIFETIME_MINUTES)
+    return get_jwt_token(user_id, expiration, token_type="access")
+
+
+def jwt_refresh_token(user_id: int) -> str:
+    expiration = timedelta(days=settings.JWT_REFRESH_TOKEN_LIFETIME_DAYS)
+    return get_jwt_token(user_id, expiration, token_type="refresh")
+
+
+def get_jwt_token(user_id, expiration, **kwargs):
+    payload = {
+        "user_id": user_id,
+        "exp": datetime.now() + expiration,
+        **kwargs,
+    }
+
+    return jwt.encode(payload, settings.SECRET_KEY, JWT_ALGORITHM)
+
+
+def parse_jwt_token(token: str) -> dict:
+    try:
+        return jwt.decode(token, settings.SECRET_KEY, JWT_ALGORITHM)
+    except jwt.DecodeError as e:
+        raise InvalidTokenError("Bad token") from e
+    except jwt.ExpiredSignature as e:
+        raise InvalidTokenError("Expired token") from e
+
+
+def validate_jwt_token(token: str, token_type: str) -> User:
+    token_dict = parse_jwt_token(token)
+
+    for key in ["user_id", "exp", "token_type"]:
+        if key not in token_dict:
+            raise InvalidTokenError("Missing key in token")
+
+    if token_dict["token_type"] != token_type:
+        raise InvalidTokenError("Invalid token type")
+
+    try:
+        return User.objects.get(id=token_dict["user_id"])
+    except User.DoesNotExist:
+        raise InvalidTokenError("Invalid user")
```

Comment:
@bzwei Consider explicitly re-raising using 'except Exception as exc' and 'raise InvalidTokenError('Invalid user') from exc' (raise-missing-from)